src/backend/api/meals.js

```diff
@@ -1,15 +1,133 @@
-const express = require("express");
+const express = require('express');
+const { andHaving } = require('../database');
 const router = express.Router();
-const knex = require("../database");
-
-router.get("/", async (request, response) => {
-  try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
-  }
+const knex = require('../database');
+
+router.post('/', async (request, response) => {
+    try {
+        // knex syntax for selecting things.
+        const insertMeal = await knex('meal').insert({
+            title: request.body.title,
+            description: request.body.description,
+            location: request.body.location,
+            when: request.body.when,
+            max_reservations: request.body.max_reservations,
+            price: request.body.price,
+            created_date: request.body.created_date,
+        });
+        response.json(insertMeal);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.put('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal')
+            .where({ id: request.params.id })
+            .update(request.body);
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.delete('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal')
+            .where({ id: request.params.id })
+            .delete(request.body);
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+router.get('/:id', async (request, response) => {
+    try {
+        const titles = await knex('meal').where({ id: request.params.id });
+        response.json(titles);
+    } catch (error) {
+        throw error;
+    }
+});
+
+//api/meals?maxPrice=90
+router.get('/', async (request, response) => {
+    try {
+        let titles = await knex('meal');
+
+        if ('maxPrice' in request.query) {
+            const maxPrice = Number(request.query.maxPrice);
+            titles = await (
+                await knex('meal')
+            ).filter((meal) => meal.price < maxPrice);
+            if (isNaN(request.query.maxPrice)) {
+                return response.send('Not a number max price');
+            } else {
+                return response.json(titles);
+            }
```

Comment:
It is better to check if the query parameter is not valid before querying the database.
That way you avoid an unnecessary query and thus, make your application respond faster.
This is a general idea to have in mind: You should try to keep your queries to the db to a minimum, cause in this case the meal app contains only a few records, but what if it grows to millions and millions? It will start to get slower..